python/dolfinx/io/utils.py

```diff
@@ -186,6 +201,29 @@ def write(self, t: float):
         def close(self):
             self._cpp_object.close()
 
+    class ADIOS2(_cpp.io.ADIOS2):
+        def __enter__(self):
+            return self
+
+        def __exit__(self, exception_type, exception_value, traceback):
+            self.close()
+
+    def write_mesh(ADIOS2: ADIOS2, mesh: Mesh) -> None:
+        """Write mesh to a file using ADIOS2"""
+        dtype = mesh.geometry.x.dtype  # type: ignore
+        if np.issubdtype(dtype, np.float32):
+            _writer = _cpp.io.write_mesh_float32
+        elif np.issubdtype(dtype, np.float64):
+            _writer = _cpp.io.write_mesh_float64
+
+        return _writer(ADIOS2, mesh._cpp_object)
+
+    def read_mesh(_ADIOS2: ADIOS2, ADIOS2: ADIOS2, comm: _MPI.Comm) -> Mesh:
```

Comment:
Not clear how or why MPI comm can be different to the one used to construct the Adios2 reader/writer.

Reply:
ADIOS2 doesn't have a public API to fetch its `MPI_Comm`.
So we have to pass the `comm` separately.